libcodechecker/libhandlers/analyze.py

```diff
@@ -177,6 +184,58 @@ def add_arguments_to_parser(parser):
                                help="File containing argument which will be "
                                     "forwarded verbatim for Clang-Tidy.")
 
+    if is_ctu_capable():
+        ctu_opts = parser.add_argument_group('cross translation unit analysis')
+        ctu_mutex_group = ctu_opts.add_mutually_exclusive_group()
+        ctu_mutex_group.add_argument('--ctu',
+                                     action='store_const',
+                                     const=[True, True],
+                                     dest='ctu_phases',
+                                     default=argparse.SUPPRESS,
+                                     help="Perform cross translation unit "
+                                          "(CTU) analysis (both collect and "
+                                          "analyze phases) using default "
+                                          "<ctu-dir> for temporary output. "
+                                          "At the end of the analysis, the "
+                                          "temporary directory is removed.")
+        ctu_opts.add_argument('--ctu-dir',
+                              metavar='<ctu-dir>',
+                              default=os.path.join(
+                                  util.get_default_workspace(),
+                                  'ctu-dir'),
+                              help="Optional parameter defines the temporary "
+                                   "directory used between CTU phases.")
+        ctu_mutex_group.add_argument('--ctu-collect-only',
```

Comment:
I really have a deep concern about having this option here, in `analyze`. `analyze` is supposed to **run analysis**, and NOT prepare and/or create some dumps that are "to be used later". This option undermines the concept of this command.

I would be more happy if at least the "collect" option would be part of `CodeChecker log` instead. But introducing a separate sub-command would be the best course?

`analyze` has a precondition, that build logs are available. (and that `clang` is, etc.)
Thus, a CTU-version of `analyze` should have the precondition that alongside the build logs, the CTU dumps are also already available.

What happens if `--ctu-collect-only` (note my other comment about negative optstrings) is stated? The collection runs, an analysis (which is the cornerstone of the `analyze` command) does not?

If we are not to introduce `CodeChecker ctu` as a separate subcommand (we shouldn't), then at least how `check` was split into `log`+`analyze` (ignore `store` for now), so should an implicit `ctu` be split into `ctu-log` and `ctu-analyze`, with the subcommands handling parts of the full stack respectively.

In my opinion, `log` should use the `func-map` binary (@gerazo I take it right that that command is needed for the "1st phase" only?) from a config (see earlier comments). It should have a `--ctu-collect-output` option, that specifies the folder into which the dumps should be put. An optional argument, which **if** specified, automatically run the dumping.

And `analyze` should get a `--use-ctu-dumps` option (or something akin), that specifies in which folder (so it's a path argument, just like `--sqlite` in `server`.) the CTU dumps are available (akin to a positional argument specifying the `build.json`s to use), and if this is given, we infer that CTU-analysis must be ran (using the given logfiles).
The alternative approach would be, mutually excluding `--use-ctu-dumps` (the opstring is up for debate!), `--ctu-on-the-fly` (or somesuch), which would trigger CTU mechanics described below.

---

And of course, the unified set of arguments could be presented on the interface of `check` and `quickcheck`, which, with proper routing, would handle CTU usage. `check` would do `log --ctu` and then `--use-ctu-dumps`; whereas `quickcheck` should do

 * either a "simple" logging and then `--ctu-on-the-fly`
 * or a `log --ctu` and then `--use-ctu-dumps`, where the CTU files (just as how the `plist`s in `quickcheck`'s case) are cleaned up later on.